src/thor/bidirectional_astar.cc

```diff
@@ -917,129 +918,188 @@ void BidirectionalAStar::SetDestination(GraphReader& graphreader, const valhalla
+    desired_paths += options.alternates();
+    // Cull alternate paths longer than maximum stretch
+    // TODO: we should skip adding the connection at all if it's greater than stretch
+    filter_alternates_by_stretch(best_connections_);
+  }
+  // For looking up edge ids on previously chosen best paths
+  std::vector<std::unordered_set<GraphId>> shared_edgeids;
+
+  // get maximum amount of sharing parameter based on origin->destination distance
+  float max_sharing =
+      allow_alternates ? get_max_sharing(options.locations(0), options.locations(1)) : 0.f;
+
+  LOG_DEBUG("Connections after stretch filter: " + std::to_string(best_connections_.size()));
+
+#ifdef LOGGING_LEVEL_TRACE
+  LOG_TRACE("CONNECTIONS FOUND " + std::to_string(best_connections_.size()));
+  for (const auto& b : best_connections_) {
+    const auto* tile = graphreader.GetGraphTile(b.edgeid);
+    auto nodes = graphreader.GetDirectedEdgeNodes(b.edgeid, tile);
+    auto sll = graphreader.GetGraphTile(nodes.first)
+                   ->node(nodes.first)
+                   ->latlng(graphreader.GetGraphTile(nodes.first)->header()->base_ll());
+    auto ell = graphreader.GetGraphTile(nodes.second)
+                   ->node(nodes.second)
+                   ->latlng(graphreader.GetGraphTile(nodes.second)->header()->base_ll());
+    printf("[[%.6f,%.6f],[%.6f,%.6f]],\n", sll.lng(), sll.lat(), ell.lng(), ell.lat());
   }
+#endif
 
-  // Reverse the list
-  std::reverse(path.begin(), path.end());
-
-  // Special case code if the last edge of the forward path is the destination edge
-  // which means we need to worry about partial distance on the edge
-  if (edgelabels_reverse_[idx2].predecessor() == kInvalidLabel) {
-    // the destination is on a different edge than origin but the forward path found it. because of
-    // that we know that the forward path did not care about partial distance along the edge. so the
-    // edge that it added was the full length (time and cost). so what we need to do is look at the
-    // second to last edge and use the reverse path (who does care about partial distance on the
-    // destination edge) to recompute the elapsed time and cost. dont forget the transition cost from
-    // the forward path
-    if (path.size() > 1) {
-      path.back().elapsed_cost.secs = path[path.size() - 2].elapsed_cost.secs +
-                                      edgelabels_reverse_[idx2].cost().secs +
-                                      edgelabels_forward_[idx1].transition_cost().secs;
-      path.back().elapsed_cost.cost = path[path.size() - 2].elapsed_cost.cost +
-                                      edgelabels_reverse_[idx2].cost().cost +
-                                      edgelabels_forward_[idx1].transition_cost().cost;
-      return paths;
+  // we quit making paths as soon as we've reached the number of paths
+  // that were requested or we run out of paths that we can actually make
+  std::vector<std::vector<PathInfo>> paths;
```

Comment:
the rest of this method looks like a big diff, but only because its in a for loop for each alternative. the guts of the for loop is pretty much the same as the old code